rasa/core/test.py

```diff
@@ -802,11 +860,21 @@ async def test(
             report_filename = os.path.join(out_directory, REPORT_STORIES_FILE)
             rasa.shared.utils.io.dump_obj_as_json_to_file(report_filename, report)
             logger.info(f"Stories report saved to {report_filename}.")
+
         else:
             report, precision, f1, accuracy = get_evaluation_metrics(
                 targets, predictions, output_dict=True
             )
 
+        evaluate_entities(
+            entity_results,
+            {"TEDPolicy"},
```

Comment:
can we extract the name from entities? I think they contain the name of the extractor

Reply:
So find all the unique extractor names in the collection of entities and evaluate them?

Reply:
I meant instead of hard coding `{"TEDPolicy"}`. How do we do it for testing nlu pipeline? We have several extractors there

Reply:
For NLU it looks at every component in the pipeline and checks if it is a subclass of `EntityExtractor`. For core we could check the config of every policy for `ENTITY_RECOGNITION=True`?

Reply:
I think you can just extract source from `EntitiesAdded` event?

Reply:
but then we would only evaluate for policies that successfully extracted an entity.

Reply:
for e2e we need to do exactly that

Reply:
I think what Joe means is that this wouldn't work in case the `TEDPolicy` isn't extracting any entity (and hence not adding an `EntitiesAdded` event)

Reply:
yes exactly, and we would want to show that the recall is 0.

Reply:
I see, yeah. Is there a way to know which algorithms are expected to extract these entities? I don't like the idea of hardcoding TED here

Reply:
Well we could look for `ENTITY_RECOGNITION` in the config of every policy in the ensemble. But that would only work if that exact config key was always used.

Reply:
maybe for simplicity, in order to unblock us, let's create a constant with a list of policies that can extract e2e entities for now?